### ISSUE TITLE
Error using Elixir 1.6.6

### ISSUE THREAD

**Issue by imetallica** [elixir-lang/elixir#7807]
When using Streams to read a file in our production server after deploying the new Elixir version with Docker, we had this error.

I haven't investigated deeper, although the stacktrace says it's a problem on core elixir.

### Environment

* Elixir 1.6.6, Erlang 21.0 
* Linux, official docker image

### Current behavior

```
** (FunctionClauseError) no function clause matching in :raw_file_io_raw.open_layer/3    
    
    The following arguments were given to :raw_file_io_raw.open_layer/3:
    
        # 1
        "/tmp/20180627.cnb"
    
        # 2 
        [:read, :read_ahead, {:read_ahead, 65536}, :binary, :binary]
    
        # 3
        [:raw, :raw]
    
    (kernel) raw_file_io_raw.erl:24: :raw_file_io_raw.open_layer/3
    (elixir) lib/file/stream.ex:78: anonymous fn/2 in Enumerable.File.Stream.reduce/3
    (elixir) lib/stream.ex:1346: anonymous fn/5 in Stream.resource/3
    (elixir) lib/stream.ex:1536: Enumerable.Stream.do_each/4
    (elixir) lib/task/supervised.ex:295: Task.Supervised.stream_reduce/7
    (elixir) lib/enum.ex:1919: Enum.reverse/1
    (elixir) lib/enum.ex:2588: Enum.to_list/1
```

**Comment by michalmuskala** [comment]
Duplicate of https://github.com/elixir-lang/elixir/issues/7771

TL;DR: `File.stream!` does not support `:raw` or `:read_ahead` modes - passing them erroneously wasn't signalled before, but now it is.
